server/api/game/controller.js

```diff
@@ -0,0 +1,124 @@
+const uuidv1 = require('uuid/v1');
+const {
+  boardCorners,
+  refreshRate,
+  getGhosts,
+} = require('./constants');
+const {
+  getRandomAdjacentAvailableCell,
+  getGridwithWeights,
+  chaseLocation,
+  initSquareGridState,
+} = require('./core');
+
+const gameState = {
+  moveGhostsCount: 0,
+  scatterGhostspath: [],
+  scatterStart: 75,
+  pacmanOne: {},
+  pacmanTwo: {},
+  gridState: [],
+};
+
+const setInitialGameState = () => {
+  const gridState = initSquareGridState();
+  const ghostsArray = getGhosts().map(([x, y, direction]) => ({
+    x, y, direction,
+  }));
+  gameState.gridState = gridState;
+  gameState.ghosts = ghostsArray;
+};
+
+const createRoom = (playerInfo, socket) => {
+  gameState.pacmanOne = playerInfo;
+  gameState.socket = socket;
+  gameState.roomId = uuidv1();
+  gameState.freeToJoin = true;
+  return gameState;
+};
+
+const currentGameState = () => {
+  const { socket, ...rest } = gameState;
+  return rest;
+};
+
+const addPositionsToArray = (arr, index) => {
+  const scatterTime = 55;
+  if (arr.length < scatterTime) {
+    arr.push(boardCorners[index]);
+    return addPositionsToArray(arr, index);
+  }
+  return arr;
+};
+
+const moveGhosts = ({ ghosts, gridState, scatterStart }) => {
+  let { moveGhostsCount, scatterGhostspath } = gameState;
+  moveGhostsCount += 1;
+  const scatterEnd = scatterStart + 55;
```

Comment:
What is 55?

Reply:
That's the duration of the scatter mode. The time for which the Ghosts spread out on the board, so for 55 * refreshrate = t sec.

Reply:
Will assign a variable for better understanding